sftp: Use MkdirAll provided by the client

fd0 opened the pull request:
<!--
Thank you very much for contributing code or documentation to restic! Please
fill out the following questions to make it easier for us to review your
changes.

You do not need to check all the boxes below all at once, feel free to take
your time and add more commits. If you're done and ready for review, please
check the last box.
-->

What is the purpose of this change? What does it change?
--------------------------------------------------------

Use the `MkdirAll()` function provided by the sftp client library instead of
our own implementation, which seems to contain a bug.

<!--
Describe the changes here, as detailed as needed.
-->

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

Closes #2518
Closes #2363 
Closes #596 

<!--
Link issues and relevant forum posts here.

If this PR resolves an issue on GitHub, use "closes #1234" so that the issue is
closed automatically when this PR is merged.
-->

Checklist
---------

- [x] I'm done, this Pull Request is ready for review

fd0 committed:
sftp: Use MkdirAll provided by the client

Closes #2518

mikl commented:
As requested, I tried building this branch on my server, resulting in this:

```
/home/mikl/restic/restic version
restic 0.9.6 (v0.9.6-39-g2cd9c7ef) compiled with go1.10.4 on linux/amd64
```

However, it does not appear to work, here’s what happens when I run it on an existing folder (created earlier by reproducing the stack byte limit):

```
/home/mikl/restic/restic -r sftp:myserver.example.org:home/restic-repo2 init
Fatal: create repository at sftp:myserver.example.org:home/restic-repo2 failed: failed to send packet: write |1: file already closed
```

And when running against a folder that did not exist, a similar error:

```
/home/mikl/restic/restic -r sftp:myserver.example.org:home/restic-repo3 init
Fatal: create repository at sftp:myserver.example.org:home/restic-repo3 failed: EOF
```

I can’t say I really understand what’s going on here, but if its of any use, I’ll be happy to furnish you with an SFTP account to test it yourself. If that’s of interest, you can write me at mikkel `[Affenschwanz]` hoegh.org.

fd0 commented:
I've sent you an email, let's get to the bottom of this! :)

fd0 committed:
Add hint abouth absolute sftp paths for Synology NAS

fd0 committed:
Add changelog file

fd0 commented:
Thanks to @mikl for providing a test account. A Synology NAS requires an absolute path when accessed via sftp. I've fixed the endless recursion and added a paragraph to the FAQ.

mikl commented on changelog/unreleased/issue-2518:
```diff
@@ -0,0 +1,16 @@
+It was found that when restic is used to store data on an sftp server on a
+Synology NAS with a relative path (one which does not start with a slash), it
+may go into an endless loop trying to create directories on the server. We've
+fixed this bug by using a function oft the sftp library instead of our own
```
"oft" should be "of"

rawtaz committed:
Minor speling corection.

rawtaz reviewed (approved):
LGTM!
